src/ast/irbuilderbpf.cpp

```diff
@@ -551,9 +553,10 @@ CallInst *IRBuilderBPF::CreateProbeReadStr(Value *ctx,
                                            const location &loc)
 {
   assert(ctx && ctx->getType() == getInt8PtrTy());
-  Constant *fn = createProbeReadStrFn(dst->getType(), src->getType(), as);
+  auto fn = createProbeReadStrFn(dst->getType(), src->getType(), as);
   auto read_fn = selectProbeReadHelper(as, true);
```

Comment:
This is kinda silly - we use createProbeReadStrFn to do the helper selection but then do it ourselves for the error message?

How about we make the `size_t size` override call the `llvm::Value * size` version, and then inline createProbeReadStrFn into the `llvm::Value size` version? It would remove the awkward tuple return too.

Reply:
Yeah, good observation. I implemented it, please check.

Reply:
Looks good, thanks!